sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpProducer.cs

```diff
@@ -465,6 +552,19 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
                     await Task.Delay(15, cancellationToken).ConfigureAwait(false);
                     MaximumMessageSize = (long)link.Settings.MaxMessageSize;
                 }
+
+                if (InitializedPartitionProperties == null)
+                {
+                    if ((ActiveFeatures & TransportProducerFeatures.IdempotentPublishing) != 0)
+                    {
+                        throw new NotImplementedException(nameof(CreateLinkAndEnsureProducerStateAsync));
```

Comment:
It would be were this not an interim state where I really just haven't started the service integration work yet.  :smile:

This will need to be implemented for the actual feature to be end-to-end complete.